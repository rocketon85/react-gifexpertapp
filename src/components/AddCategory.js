import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handlerChange = (e) => {
        setInputValue(e.target.value);
    }

    const handlerSubmit = (e) => {
        e.preventDefault();

        if (setCategories(inputValue)) { //setCategories(cats => [...cats, inputValue] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handlerSubmit}>
            <p> {inputValue} </p>
            <input 
                type="text"
                value={inputValue}
                onChange={ handlerChange }
            />
        </form>
    )
}

AddCategory.propTypes  = {
    setCategories: PropTypes.func.isRequired
}