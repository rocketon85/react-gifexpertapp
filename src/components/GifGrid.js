import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({cat}) => {
   
    const {data:images, loading} = useFetchGifs(cat);
  

    return (
        <>
            <h3>{cat}</h3>

            {loading && <p>Loading</p>}

            <div className='card-grid animate__animated animate__bounce'>
                
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id}
                            {...img} 
                        />))
                }
            
            </div>
        </>
    )
}


GifGrid.propTypes  = {
    cat: PropTypes.string.isRequired
}