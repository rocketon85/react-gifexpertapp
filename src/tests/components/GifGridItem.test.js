import React from 'react';
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('pruebas GrifGridItem', () => {

    test('debe mostrar el componente correctamente ', () => {
        const wrapper = shallow(<GifGridItem />);

        expect(wrapper).toMatchSnapshot();
    })
    
})
