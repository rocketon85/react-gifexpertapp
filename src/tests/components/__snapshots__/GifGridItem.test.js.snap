// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas GrifGridItem debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__shakeX"
>
  <img />
  <p>
            
  </p>
</div>
`;
